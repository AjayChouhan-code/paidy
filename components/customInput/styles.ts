import { Colors } from "@theme";
import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    marginVertical: 10,
    width: "100%",
  },
  label: {
    fontSize: 16,
    fontWeight: "bold",
    color: Colors.dark_grey,
    marginBottom: 5,
  },
  input: {
    borderWidth: 1,
    borderColor: Colors.light_grey,
    borderRadius: 5,
    padding: 10,
    fontSize: 14,
    color: Colors.dark_grey,
  },
  errorText: {
    color: Colors.red,
    fontSize: 12,
    marginTop: 5,
  },
});

export default styles;
