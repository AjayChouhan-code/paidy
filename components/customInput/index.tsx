import React from "react";
import { View, Text, TextInput } from "react-native";

import { CustomInputProps } from "./types";
import styles from "./styles";

export const CustomInput: React.FC<CustomInputProps> = ({
  label,
  style,
  error,
  ...textInputProps
}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.label}>{label}</Text>
      <TextInput style={[styles.input, style]} {...textInputProps} />
      {error && <Text style={styles.errorText}>{error}</Text>}
    </View>
  );
};
