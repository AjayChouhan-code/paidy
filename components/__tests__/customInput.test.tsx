import React from "react";
import renderer from "react-test-renderer";
import { CustomInput } from "../customInput";

describe("CustomInput Component", () => {
  it("renders correctly with required props", () => {
    const tree = renderer
      .create(
        <CustomInput label="Username" placeholder="Enter your username" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("applies custom styles", () => {
    const tree = renderer
      .create(
        <CustomInput
          label="Password"
          placeholder="Enter your password"
          style={{ borderColor: "red" }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("displays an error message", () => {
    const tree = renderer
      .create(
        <CustomInput
          label="Email"
          placeholder="Enter your email"
          error="Invalid email address"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes props to TextInput", () => {
    const tree = renderer
      .create(
        <CustomInput
          label="Username"
          placeholder="Enter your username"
          keyboardType="email-address"
          maxLength={20}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
