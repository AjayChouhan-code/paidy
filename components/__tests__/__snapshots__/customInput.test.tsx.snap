// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput Component applies custom styles 1`] = `
<View
  style={
    {
      "marginVertical": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Password
  </Text>
  <TextInput
    placeholder="Enter your password"
    style={
      [
        {
          "borderColor": "#ccc",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#333",
          "fontSize": 14,
          "padding": 10,
        },
        {
          "borderColor": "red",
        },
      ]
    }
  />
</View>
`;

exports[`CustomInput Component displays an error message 1`] = `
<View
  style={
    {
      "marginVertical": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Email
  </Text>
  <TextInput
    placeholder="Enter your email"
    style={
      [
        {
          "borderColor": "#ccc",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#333",
          "fontSize": 14,
          "padding": 10,
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      {
        "color": "red",
        "fontSize": 12,
        "marginTop": 5,
      }
    }
  >
    Invalid email address
  </Text>
</View>
`;

exports[`CustomInput Component passes props to TextInput 1`] = `
<View
  style={
    {
      "marginVertical": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Username
  </Text>
  <TextInput
    keyboardType="email-address"
    maxLength={20}
    placeholder="Enter your username"
    style={
      [
        {
          "borderColor": "#ccc",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#333",
          "fontSize": 14,
          "padding": 10,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`CustomInput Component renders correctly with required props 1`] = `
<View
  style={
    {
      "marginVertical": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Username
  </Text>
  <TextInput
    placeholder="Enter your username"
    style={
      [
        {
          "borderColor": "#ccc",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#333",
          "fontSize": 14,
          "padding": 10,
        },
        undefined,
      ]
    }
  />
</View>
`;
